src/phoenix/server/api/schema.py

```diff
@@ -229,17 +245,19 @@ def hdbscan_clustering(
 @strawberry.type
 class Mutation(ExportEventsMutation):
     @strawberry.mutation
-    def delete_project(self, info: Info[Context, None], id: GlobalID) -> Query:
-        if (traces := info.context.traces) is not None:
-            node_id = from_global_id_with_expected_type(str(id), "Project")
-            traces.archive_project(node_id)
-        return Query()
-
-    @strawberry.mutation
-    def archive_project(self, info: Info[Context, None], id: GlobalID) -> Query:
-        if (traces := info.context.traces) is not None:
-            node_id = from_global_id_with_expected_type(str(id), "Project")
-            traces.archive_project(node_id)
+    async def delete_project(self, info: Info[Context, None], id: GlobalID) -> Query:
+        node_id = from_global_id_with_expected_type(str(id), "Project")
+        async with info.context.db() as session:
+            project = await session.scalar(
+                select(models.Project)
+                .where(models.Project.id == node_id)
+                .options(load_only(models.Project.name))
+            )
+            if project is None:
+                raise ValueError(f"Unknown project: {id}")
```

Comment:
Suggestion: Can save a line or two with `if (project :=...`

Reply:
the number of lines is the same due to formatting.

<img width="316" alt="Screenshot 2024-04-12 at 10 38 18 AM" src="https://github.com/Arize-ai/phoenix/assets/80478925/4b9a9b34-2678-4dfd-99ab-4b8f30b58b21">